go-apps/meep-ams/server/ams.go

```diff
@@ -627,34 +659,34 @@ func mec011AppTerminationPost(w http.ResponseWriter, r *http.Request) {
 	}
 
 	go func() {
-		//delete any registration it made
-		// cannot unsubscribe otherwise, the app-enablement server fails when receiving the confirm_terminate
-		// since it believes it never registered
-		//_ = unsubscribeAppTermination(serviceAppInstanceId)
+		// Deregister service
 		_ = deregisterService(serviceAppInstanceId, appEnablementServiceId)
 
-		//send scenario update with a deletion
-		var event scc.Event
-		var eventScenarioUpdate scc.EventScenarioUpdate
-		var process scc.Process
-		var nodeDataUnion scc.NodeDataUnion
-		var node scc.ScenarioNode
-
-		process.Name = instanceName
-		process.Type_ = "EDGE-APP"
-
-		nodeDataUnion.Process = &process
-
-		node.Type_ = "EDGE-APP"
-		node.Parent = mepName
-		node.NodeDataUnion = &nodeDataUnion
-
-		eventScenarioUpdate.Node = &node
-		eventScenarioUpdate.Action = "REMOVE"
+		// Delete subscriptions
+		_ = unsubscribeAppTermination(serviceAppInstanceId, appTermSubId)
 
-		event.EventScenarioUpdate = &eventScenarioUpdate
-		event.Type_ = "SCENARIO-UPDATE"
+		// Confirm App termination if necessary
+		if sendAppTerminationWhenDone {
+			_ = sendTerminationConfirmation(serviceAppInstanceId)
+		}
 
+		// Remove node from active scenario
```

Comment:
When we execute AMS in 'global' mode, this would cause an unnecessary event.
To be on the safe side, we should just issue it if necessary (e.g. distributed mode)

Reply:
Implementation reworked in PR #327. Event is now triggered from Sandbox Controller.